### ISSUE TITLE
High CPU usage on Kali Linux in a Virtual Machine

### ISSUE THREAD

**Issue by 0xsnowcloud** [warpdotdev/Warp#4710]
### Discord username (optional)

0xsnowcloud

### Describe the bug

On Kali Linux in a Virtual Box VM, Warp uses a high amount of CPU. It also has noticeable lag when typing in the command prompt. The CPU usage on the VM drops to normal when the warp terminal looses mouse focus. 

### To reproduce

1. Download and install a Kali Linux Virtual Box VM (https://www.kali.org/get-kali/#kali-virtual-machines)
2. Install Warp via the .deb package
3. Open Task manager and observe the CPU usage
4. Open Warp via the application menu
5. Review the elevated CPU usage while the Warp terminal has "focus"

### Expected behavior

On Arch Linux, Warp uses .03% of the CPU as opposed to 70%+ CPU when in the VM

### Screenshots

![VirtualBox_kali-linux-2023](https://github.com/warpdotdev/Warp/assets/10764378/fdb1088d-e5ad-4b98-90cb-f9c795b291c2)

### Operating system

Linux

### Operating system and version

Kali GNU/Linux Rolling 2024.1

### Shell Version

zsh 5.9 (x86_64-debian-linux-gnu)

### Current Warp version

v0.2024.04.02.08.02.stable_01

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Additional context

I am currently running Warp with X11

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hi @0xsnowcloud Since Warp is GPU accelerated, on virtual machines the GPU is emulated by the CPU so you should see higher usage on the CPU in VMs. As far as I'm aware Warp isn't optimized to make use of 3D acceleration features on VM hypervisors like virtualbox/VMware, even if enabled.

That being said, we've added a list of workarounds here: https://docs.warp.dev/help/known-issues#linux-workarounds

Also, make sure to [download the latest version](https://warp.dev/linux-terminal) as we release fixes every week. https://docs.warp.dev/getting-started/changelog Some of those may help with the issue.

**Comment by 0xsnowcloud** [comment]
Hey @dannyneira thanks for the reply. I have tried the list of workarounds with no changes in results and I am also using the latest version of Warp. I do acknowledge that Warp is GPU accelerated and I am using a Virtual Machine, so that is part of the reason why I am seeing higher CPU usage, however I have noticed some additional "suspicious" behaviors.

I have noticed that the CPU usage gets elevated under two conditions in my Kali VM:
1. When no active processes are running in any terminal windows (in effect, Warp is "
idle")
2. When clicking anywhere in the Warp window pane, including rapidly clicking.

In summary, when using Warp and running tasks, the CPU usage actually goes down. When allowing the terminal window to "idle" or rapidly clicking around the window the CPU usage spikes.

Hope this helps!

**Comment by dannyneira** [comment]
Thanks for the extra contexts on the issue.well investigate this further and report back if there are any changes
